src/cim/components/modals/AddBmcHostYamlForm.tsx

```diff
@@ -0,0 +1,302 @@
+        online: false,
+      },
+    };
+
+    const host2 = {
+      apiVersion: 'metal3.io/v1alpha1',
+      kind: 'BareMetalHost',
+      metadata: {
+        annotations: {
+          'bmac.agent-install.openshift.io/hostname': '<<the name of hostname>>',
+          'inspect.metal3.io': 'disabled',
+        },
+        labels: {
+          'infraenvs.agent-install.openshift.io':
+            '<<the name of the infra env in which this host will be>>',
+        },
+        name: '<<name of bare metal host>>',
+        namespace: '<<name of namespace>>',
+      },
+      spec: {
+        automatedCleaningMode: 'disabled',
+        bmc: {
+          address: '<<Baseboard Management Controller Address>>',
+          credentialsName: '<<name of credentials indicate in 7th line>>',
+          disableCertificateVerification: true,
+        },
+        bootMACAddress:
+          "<<The MAC address of the host's network connected NIC that wll be used to provision the host.>>",
+        online: false,
+      },
+    };
+    const yaml = [credentials, host1, host2].map((doc) => jsYaml.dump(doc)).join('---\n');
+    const blob = new Blob([yaml], { type: 'text/plain;charset=utf-8' });
+    saveAs(blob, 'bmc_hosts_template.yaml');
+  };
+
+  async function importBMAsYAML() {
+    // eslint-disable-next-line
+    return new Promise<any[] | string[]>((resolve, reject) => {
+      setTextOpenFile('Opening file');
+      const input = document.createElement('input');
+      input.setAttribute('id', 'importBMAsYaml');
+      input.type = 'file';
+      input.accept = '.yaml';
+      input.style.visibility = 'hidden';
+      const body = document.querySelector('body');
+      body?.appendChild(input);
+      input.onchange = (e: Event) => {
+        input.remove();
+        const target = e.target as HTMLInputElement;
+        const file: File = (target.files as FileList)[0];
+        let maxSize = 2147483648; //2Gb
+        maxSize = 529145600; //600 Mb per provar
+        if (file.size > maxSize) {
+          resolve(['', file.name, 'Error1']);
+        } else {
+          const reader = new FileReader();
+          reader.readAsText(file, 'UTF-8');
+          reader.onload = async (readerEvent) => {
+            const content = readerEvent.target?.result;
+            // parse yaml
+            if (typeof content === 'string') {
+              await new Promise((r) => setTimeout(r, 100));
+              let yamlContent;
+              try {
+                yamlContent = jsYaml.loadAll(content);
+                setTextOpenFile('Open file');
+                resolve([yamlContent, file.name]);
+              } catch (err) {
+                setTextOpenFile('Open file');
+                reject(err);
+              }
+            }
+          };
+        }
+      };
+      input.click();
+    });
+  }
+
+  const getTextError = (error: string) => {
+    switch (error) {
+      case 'Error1':
```

Comment:
why `Error1`/ `Error2` when you can use the error texts directly  ? `The file is too big. Upload a file up to 2 GB.` / `File is not structured correctly. Use the template to use the right file structure.`